src/snowflake/snowpark/dataframe.py

```diff
@@ -4782,8 +4789,12 @@ def cache_result(
             f'"{random_name_for_temp_object(TempObjectType.TABLE)}"'
         )
 
+        # TODO: Clarify whether cache_result() is an Eval or not. Currently, treat as Assign.
```

Comment:
Let's confer in the next meeting with us and Ovidiu. I'm really into being super explicit about side effects.

Reply:
Yes, I came across this while I was fixing something. Added a comment for this reason, because myself I am unsure how we would want to treat this here (i.e., issue an explicit Eval or not)